core/src/main/java/edu/wpi/grip/core/operations/networktables/NTPublishOperation.java

```diff
@@ -0,0 +1,140 @@
+package edu.wpi.grip.core.operations.networktables;
+
+import com.google.common.base.Throwables;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.first.wpilibj.networktables.NetworkTable;
+import edu.wpi.first.wpilibj.tables.ITable;
+import edu.wpi.grip.core.*;
+import edu.wpi.grip.core.events.ProjectSettingsChangedEvent;
+
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * An operation that publishes any type that implements {@link NTPublishable} to NetworkTables.
+ * <p>
+ * To be publishable, a type should have one or more accessor methods annotated with {@link NTValue}.  This is done
+ * with annotations instead of methods
+ */
+public class NTPublishOperation<T extends NTPublishable> implements Operation {
+
+    static {
+        NetworkTable.setClientMode();
+    }
+
+    private final NetworkTable table;
+    private final Class<T> type;
+    private final List<Method> ntValueMethods = new ArrayList<>();
+
+    public NTPublishOperation(Class<T> type) {
+        this.table = NetworkTable.getTable("GRIP");
+        this.type = checkNotNull(type, "Type was null");
+
+        // Any accessor method with an @NTValue annotation can be published to NetworkTables.
+        for (Method method : type.getDeclaredMethods()) {
+            if (method.getAnnotation(NTValue.class) != null) {
+                if (method.getParameters().length > 0) {
+                    throw new IllegalArgumentException("@NTValue method must have 0 parameters: " + method);
+                }
+
+                ntValueMethods.add(method);
+            }
+        }
+    }
+
+    /**
+     * Change the server address according to the team number preference.  The NetworkTables server runs on the robot,
+     * which as the hostname that looks like "roborio-frc-xxxx.local".
+     */
+    @Subscribe
+    public void updateSettings(ProjectSettingsChangedEvent event) {
```

Comment:
This is actually a bad idea. What happens when we add more NTPublishOperation for classes other than `ContoursReport`?